app/src/main/java/org/openobservatory/ooniprobe/fragment/ResultListFragment.java

```diff
@@ -53,125 +62,207 @@
+            default:
+                return super.onOptionsItemSelected(item);
+        }
+    }
+
+    @OnItemSelected(R.id.filterTests)
+    void queryList() {
+        if (((Application) getActivity().getApplication()).getPreferenceManager().isManualUploadResults() &&
+                SQLite.selectCountOf().from(Measurement.class).where(
+                        Measurement_Table.is_uploaded.eq(false),
+                        OperatorGroup.clause().
+                                or(Measurement_Table.is_failed.eq(false))
+                                .or(Measurement_Table.report_id.isNull())
+                ).longValue() != 0
+        )
+            snackbar.show();
+        else
+            snackbar.dismiss();
+        HashSet<Integer> set = new HashSet<>();
+        items.clear();
+        ArrayList<SQLOperator> where = new ArrayList<>();
+        String filter = getResources().getStringArray(R.array.filterTestValues)[filterTests.getSelectedItemPosition()];
+        if (!filter.isEmpty())
+            where.add(Result_Table.test_group_name.is(filter));
+        List<Result> list = SQLite.select().from(Result.class).where(where.toArray(new SQLOperator[0]))
+                .orderBy(Result_Table.start_time, false).queryList();
+        if (list.isEmpty()) {
+            emptyState.setVisibility(View.VISIBLE);
+            recycler.setVisibility(View.GONE);
+        } else {
+            emptyState.setVisibility(View.GONE);
+            recycler.setVisibility(View.VISIBLE);
+            for (Result result : list) {
+                Calendar c = Calendar.getInstance();
+                c.setTime(result.start_time);
+                int key = c.get(Calendar.YEAR) * 100 + c.get(Calendar.MONTH);
+                if (!set.contains(key)) {
+                    items.add(new DateItem(result.start_time));
+                    set.add(key);
+                }
+                switch (result.test_group_name) {
+                    case WebsitesSuite.NAME:
+                        items.add(new WebsiteItem(result, this, this));
+                        break;
+                    case InstantMessagingSuite.NAME:
+                        items.add(new InstantMessagingItem(result, this, this));
+                        break;
+                    case MiddleBoxesSuite.NAME:
+                        items.add(new MiddleboxesItem(result, this, this));
+                        break;
+                    case PerformanceSuite.NAME:
+                        items.add(new PerformanceItem(result, this, this));
+                        break;
+                }
+            }
+            adapter.notifyTypesChanged();
+        }
+    }
+
+    @Override
+    public void onClick(View v) {
+        Result result = (Result) v.getTag();
+        ActivityCompat.startActivity(getActivity(), ResultDetailActivity.newIntent(getActivity(), result.id), null /* ActivityOptionsCompat.makeClipRevealAnimation(v, 0, 0, v.getWidth(), v.getHeight()).toBundle()*/);
```

Comment:
Why there is a comment here?

Reply:
is only a placeholder for removed transition